frontend/src/pages/user/UserAnswerPage.js

```diff
@@ -39,9 +42,46 @@ export default function UserAnswerPage() {
   const lastPage =
     questions && Array.isArray(questions) ? questions.length + 1 : 0;
 
+  const totalQuestion =
+    questions && Array.isArray(questions) ? questions.length : 0;
+  const passingScore = 75;
+
+  useEffect(() => {
+    setFailed(scorePercentage < passingScore ? true : false);
+    setPassed(scorePercentage >= passingScore ? true : false);
+  }, [scorePercentage]);
+
+  useEffect(() => {
+    const data = {
+      score,
+      totalQuestion,
+    };
+    const calculate = async (data) => {
+      try {
+        const result = await calculateScore(data);
+        setScorePercentage(result.result);
+      } catch (error) {
+        console.log(error);
+      }
+    };
+    calculate(data);
+  }, [isDone]);
+
+  const handleAnswer = (question, answer, correctAnswer) => {
+    setCurrentPage(currentPage + 1);
+    setAnswers(answers.concat([answer]));
+    setIsCorrect(isCorrect.concat([answer === correctAnswer ? true : false]));
```

Comment:
You still calculating in frontend if its correct or not. Again, just store all answers and let BE do the checking and calculation.